pkg/template/compiled_template_error.go

```diff
@@ -204,6 +204,26 @@ func (CompiledTemplateMultiError) hintMsg(errMsg string) string {
 		hintMsg = "use 'None' instead of 'none' to indicate no value"
 	case "unhandled index operation struct[string]":
 		hintMsg = "use getattr(...) to access struct field programmatically"
+	case "unknown binary op: bool | bool":
+		hintMsg = "use 'or' instead of '|' to combine boolean expressions"
+	case "unknown binary op: bool & bool":
+		hintMsg = "use 'and' instead of '&' to combine boolean expressions"
```

Comment:
👍🏻  All the hints you're adding here are written in the same voice (with the same typography) as the existing text.